src/secrets-kubernetes/src/lib.rs

```diff
@@ -53,29 +58,56 @@ impl KubernetesSecretsController {
             },
         };
         let client = Client::try_from(kubeconfig)?;
-        let secret_api: Api<Secret> = Api::default_namespaced(client);
-
-        let secret = KubernetesSecretsController::make_secret_with_name(SECRET_NAME.to_string());
-        match secret_api.create(&PostParams::default(), &secret).await {
-            Ok(_) => Ok(()),
-            Err(kube::Error::Api(e)) if e.code == 409 => {
-                info!("Secret {} already exists", SECRET_NAME.to_string());
-                Ok(())
-            }
-            Err(e) => {
-                error!("creating secret failed: {}", e);
-                Err(e)
-            }
-        }?;
+        let secret_api: Api<Secret> = Api::default_namespaced(client.clone());
+        let pod_api: Api<Pod> = Api::default_namespaced(client);
+
+        // ensure that the secret has been created in this environment
+        secret_api.get(&*config.user_defined_secret).await?;
+
+        if !Path::new(&config.user_defined_secret_mount_path).is_dir() {
+            bail!(
+                "Configured secrets location could not be found on filesystem: ({})",
+                config.user_defined_secret_mount_path
+            );
+        }
+
+        Ok(KubernetesSecretsController {
+            secret_api,
+            pod_api,
+            config,
+        })
+    }
+
+    async fn trigger_resync(&mut self) -> Result<(), Error> {
+        let mut coordd: Pod = self.pod_api.get(&*self.config.coordd_pod_name).await?;
+        coordd.annotations_mut().insert(
+            String::from(POD_ANNOTATION),
+            format!("{:x}", random::<u64>()),
+        );
+
+        // Managed_fields can not be present in the object that is being patched.
+        // if present, they lead to an 'metadata.managedFields must be nil' error
+        coordd.metadata.managed_fields = None;
```

Comment:
Rather than fetching the current pod spec and re-applying it with the modification (which looks racy!), I think you can just construct a `Pod::default()` and then add the annotation to that?

Reply:
I think it checks the resource version when applying with one, but don't quote me on that. I have to look it up every time.

I agree, though, it may be better to use `Pod::default()`, since that won't take ownership of EVERY field. As currently written, this may fight over ownership of the fields with the environment-controller.

I'm not a huge fan of the pod modifying itself in general. I won't block this PR for that, though.